src/IO/configuration_JSON.cpp

```diff
@@ -246,6 +282,43 @@ void writeConfiguration(const std::string &filename, const Config *rootConfig, b
     std::fclose(fp);
 }
 
+/*!
+    Write a bitpit::Config tree contents to json formatted c++ string.
+
+    NOTE: JSON root document object does not have a key name unlike XML.
+
+    Tree contents will be written to a plain c++ string, with UTF-8 standard encoding, and 
+    appended to any previous content of the source string.
+    
+    \param[in] source string to write JSON contents.
+    \param[in] rootConfig pointer to the Config tree to be written on target string
+*/
+void writeBufferConfiguration(std::string &source, const Config *rootConfig)
```

Comment:
Arguments that are modified by the function are usually passed by pointer at the end of the argument list.